js/src/edit-block/components/front-end-preview.js

```diff
@@ -25,47 +34,63 @@ import { getBlock, getBlockNameWithNameSpace } from '../helpers';
  * So there can be an error in <ServerSideRender>
  * if it passes attributes that aren't yet saved.
  *
+ * @param {FrontEndPreviewProps} props
  * @return {React.ReactElement} The front-end preview.
  */
-const FrontEndPreview = () => {
+const FrontEndPreview = ( { setEditorMode } ) => {
 	const currentPost = useSelect( ( select ) => select( 'core/editor' ).getCurrentPost() );
 	const isPostNew = useSelect( ( select ) => select( 'core/editor' ).isEditedPostNew() );
 	const isPostDirty = useSelect( ( select ) => select( 'core/editor' ).isEditedPostDirty() );
+	const { getFields } = useField();
 
 	const fullBlock = getBlock( currentPost?.content );
 	const blockNameWithNameSpace = getBlockNameWithNameSpace( fullBlock );
 	const block = fullBlock[ blockNameWithNameSpace ] || {};
+	const fields = getFields();
+
+	if ( ! getFieldsAsArray( fields ).length ) {
+		return (
+			<PreviewNotice>
+				<button
+					className="underline"
+					onClick={ () => setEditorMode( BUILDER_EDITING_MODE ) }
+				>
+					{ __( 'Builder', 'genesis-custom-blocks' ) }
+				</button>
+			</PreviewNotice>
+		);
+	}
+
+	if ( ! block.previewAttributes ) {
+		return (
+			<PreviewNotice>
+				<button
+					className="underline"
+					onClick={ () => setEditorMode( EDITOR_PREVIEW_EDITING_MODE ) }
+				>
+					{ __( 'Editor Preview', 'genesis-custom-blocks' ) }
+				</button>
+			</PreviewNotice>
+		);
+	}
 
-	// There's nothing entered or saved, so <ServerSideRender> would have an error.
-	if ( isPostNew && ! isPostDirty ) {
+	if ( isPostNew && isPostDirty ) {
 		return (
-			<p className="mt-4">
-				{ __( 'Please edit the block to preview it.', 'genesis-custom-blocks' ) }
-			</p>
+			<div className="mt-4 flex flex-row items-center">
+				{ __( 'Please save your block to preview it:', 'genesis-custom-blocks' ) }
+				&nbsp;
+				<PostSavedState />
+			</div>
 		);
 	}
 
 	return (
-		<>
-			{
-				isPostNew && isPostDirty
-					? (
-						<div className="mt-4 flex flex-row items-center">
-							{ __( 'Please save your block to preview it:', 'genesis-custom-blocks' ) }
-							&nbsp;
-							<PostSavedState />
-						</div>
-					)
-					: (
-						<ServerSideRender
```

Comment:
This file mainly works the same, this just moves the notices higher in the file, and slightly changes them.